### ISSUE TITLE
Inconsistent analysis result of same sample

### ISSUE THREAD

**Issue by falcorocks** [cuckoosandbox/cuckoo#2108]
Hi guys, I have analysed 40000 samples on the latest version of cuckoo (2.0.5).
Most of them look good. Some though have the following problem:

- report score is 0/10
- hash is strongly malicious on VT
- checking report it seems nothing really was executed, but also no error is reported

When submitted again, the malware is actually executed and the report score is above 0.

In other words, I'm struggling with inconsistent analysis result.
I understand the score is just a beta, but clearly something is very wrong...

Is this normal?
How would you take care of this? I am thinking of querying elasticsearch for all hashes with a 0/10 and run them again with a bash+api script, but it is far from ideal

**Comment by doomedraven** [comment]
`report score is 0/10` -> `cuckoo comunity` missed?

**Comment by falcorocks** [comment]
Nope, cuckoo community is there. Roughly 10-20% of analysis have a 0/10 but the others are fine

**Comment by doomedraven** [comment]
that you have something wrong, do you see the correct behavior in behavior tab?

**Comment by falcorocks** [comment]
No, 0/10 analysis have nothing in the behavioural part of the web report.
Let me be clear, when I re-run the analysis it works fine... so it's something very inconsistent...

**Comment by falcorocks** [comment]
So I managed to get to the machine running cuckoo and I was presented with this:
![cancer3](https://user-images.githubusercontent.com/14293929/35808278-958741e6-0a85-11e8-812b-a29c8d13be7f.jpeg)

**Comment by falcorocks** [comment]
it may not be clear from the screenshot, but there are like hundreds of those popups

**Comment by doomedraven** [comment]
google the result error, it looks like you have problems with vbox not cuckoo

**Comment by falcorocks** [comment]
Some more information:

- that error in the screenshot is indeed related to virtualbox, basically when virtual machines hang for whatever reason they will give that error.

- there were only ~80 of these popup dialogs

- 14680 items over 25932 total have a suspiciously low 0/10 score, well above my 10-20% initial figure.

My solution now:

- reduced the number of vms that can run in parallel from 10 to 5

- reduced the number of vms that can start in parallel from 10 to 5

- i wrote a script to re analyse the 0/10 analysis with `curl http://localhost:8090/tasks/rereport/x`

Suggestions? I find it hard to believe that it is cpu/memory issue as the machine is more than powerful enough to start 10 vms together without breaking a sweat.
May it have to do with virtualbox not being that stable itself...?

**Comment by doomedraven** [comment]
use kvm, vbox is vbox :D

**Comment by falcorocks** [comment]
@doomedraven do you really have better results with kvm? I might consider the switch if you make a compelling enough case..

**Comment by doomedraven** [comment]
for me yes

**Comment by rstgermain** [comment]
+1 for KVM. Fixed all of our issues related to the virtual machines.

**Comment by doomedraven** [comment]
just in case check :) http://www.doomedraven.com/2016/05/kvm.html#modifying-kvm-qemu-kvm-settings-for-malware-analysis

**Comment by falcorocks** [comment]
@rstgermain thanks for the feedback!
@doomedraven dude that looks amazing! Thank you!
It should be linked in the cuckoo documentation!

**Comment by doomedraven** [comment]
you are welcome ;)
